clippy_lints/src/needless_pass_by_value.rs

```diff
@@ -323,6 +319,17 @@ impl<'a, 'tcx> LateLintPass<'a, 'tcx> for NeedlessPassByValue {
     }
 }
 
+/// Functions marked with these attributes must have the exact signature.
+fn requires_exact_signature(attrs: &[Attribute]) -> bool {
+    attrs.iter().any(|attr| {
+        attr.meta_item_list().is_some() && attr.name() == "proc_macro_derive"
+            || attr.meta_item_list().is_none()
+                && ["proc_macro", "proc_macro_attribute"]
```

Comment:
I think we could just add `proc_macro_derive` to this list and scrap the special code above.